packages/test/local-server-stress-tests/src/localServerStressHarness.ts

```diff
@@ -774,102 +787,49 @@ async function loadClient(
-		};
-		c.container.once("closed", rejector);
-		c.container.once("disposed", rejector);
-		cleanUps.push(() => {
-			c.container.off("closed", rejector);
-			c.container.off("disposed", rejector);
-		});
-	}
-	try {
-		await Promise.all(
-			connectedClients.map(async (c) =>
-				timeoutPromise(
-					(resolve, reject) => {
-						const entry = closeOrDispose.get(c) ?? { rejects: [] };
-						closeOrDispose.set(c, entry);
-						if (entry.error) {
-							reject(entry.error);
-						} else if (c.container.connectionState !== ConnectionState.Connected) {
-							c.container.once("connected", () => resolve());
-							entry.rejects.push(reject);
-						} else {
-							resolve();
-						}
-					},
-					{
-						errorMsg: `Timed out waiting for client to connect ${c.tag}`,
-					},
-				),
-			),
-		);
-
-		await Promise.all(
-			connectedClients.map(async (c) =>
-				timeoutPromise(
-					(resolve, reject) => {
-						const entry = closeOrDispose.get(c) ?? { rejects: [] };
-						closeOrDispose.set(c, entry);
-						if (entry.error) {
-							reject(entry.error);
-						} else if (c.container.isDirty) {
-							c.container.once("saved", () => resolve());
-							entry.rejects.push(reject);
-						} else {
-							resolve();
-						}
-					},
-					{
-						errorMsg: `Timed out waiting for client to save ${c.tag}`,
-					},
-				),
-			),
-		);
-		const maxSeq = Math.max(
-			...connectedClients.map((c) => c.container.deltaManager.lastKnownSeqNumber),
-		);
-
-		const makeOpHandler = (c: Client, resolve: () => void, reject: (error) => void) => {
-			const entry = closeOrDispose.get(c) ?? { rejects: [] };
-			closeOrDispose.set(c, entry);
-			if (entry.error) {
-				reject(entry.error);
-			} else if (c.container.deltaManager.lastKnownSeqNumber < maxSeq) {
-				const handler = (msg) => {
-					if (msg.sequenceNumber >= maxSeq) {
-						c.container.off("op", handler);
-						resolve();
-					}
-				};
-				c.container.on("op", handler);
-				entry.rejects.push(reject);
-			} else {
+	return timeoutPromise((resolve, reject) => {
+		const handler = (error?) => {
+			if (connectedClients.some((c) => c.container.closed || c.container.disposed === true)) {
+				reject(error);
+				off();
+				return;
+			}
+			if (
+				connectedClients.every(
+					(c) =>
+						c.container.connectionState === ConnectionState.Connected &&
+						c.container.isDirty === false &&
+						c.container.deltaManager.lastSequenceNumber ===
+							connectedClients[0].container.deltaManager.lastSequenceNumber,
+				)
+			) {
 				resolve();
+				off();
 			}
 		};
-		await Promise.all(
-			connectedClients.map(async (c) =>
-				timeoutPromise((resolve, reject) => makeOpHandler(c, resolve, reject), {
-					errorMsg: `Timed out waiting for client to catch up: ${c.tag} seq: ${maxSeq}`,
-				}),
-			),
-		);
-	} finally {
-		cleanUps.forEach((f) => f());
-	}
+		const timeout = setInterval(() => {
+			// just an easy place to place a breakpoint
```

Comment:
Not sure when we'd want to debug here v. just putting a breakpoint in `handler`.  If nothing has changed in that first second... then this call to `handler` will just be a no-op.

Reply:
it lets you detect hangs. one of the bugs caused synchronization to hang due to a bad dirty state. i just left it in, as this is the place most likely to hang when debugging.